test/integration/odcds_integration_test.cc

```diff
@@ -0,0 +1,415 @@
+#include <string>
+
+#include "envoy/common/platform.h"
+#include "envoy/config/cluster/v3/cluster.pb.h"
+
+#include "source/common/common/fmt.h"
+#include "source/common/common/macros.h"
+
+#include "test/common/grpc/grpc_client_integration.h"
+#include "test/integration/ads_integration.h"
+#include "test/integration/fake_upstream.h"
+#include "test/integration/http_integration.h"
+#include "test/test_common/resources.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+const std::string& config() {
+  CONSTRUCT_ON_FIRST_USE(std::string, fmt::format(R"EOF(
+admin:
```

Comment:
Why isn't this using the standard utilities in `test/config/utility.h` at least for baseline? I'm not a huge fan of the boiler plate in these copy+paste YAMLs @adisuissa as well.

Reply:
It seems most of the config is shared with the default integration test config, so I'd suggest using: `config_helper_` here.

Reference code:
https://github.com/envoyproxy/envoy/blob/824032e93ef295e22086f465d4c60a3aeea4ae13/test/integration/http_typed_per_filter_config_test.cc#L16-L32

Reply:
Yes, definitely. We want to avoid depending on specific extensions inside `test/integration`. Thanks for spotting that.